libbitcoin/src/signer.rs

```diff
@@ -102,6 +102,7 @@ impl string_result_t {
     pub fn success(data: impl ToString) -> string_result_t {
         let (code, details) = match CString::new(data.to_string()) {
             Ok(s) => (error_t::success, result_details_t { data: s.into_raw() }),
+            #[allow(clippy::needless_borrow)]
```

Comment:
Why just not remove the borrow here?

Reply:
It's because the CString NulError is of a different type, and there was some case where Clippy couldn't make sense of things.

Works:
```rust
#[allow(clippy::needless_borrow)]
Err(err) => (error_t::invalid_result_data, (&err).into()),
```

Does not work:
```rust
Err(err) => (error_t::invalid_result_data, err.into()),
```
Why:
```
the trait bound `signer::result_details_t: std::convert::From<std::ffi::NulError>` is not satisfied
the following implementations were found:
  <signer::result_details_t as std::convert::From<&E>>
required because of the requirements on the impl of `std::convert::Into<signer::result_details_t>` for `std::ffi::NulError`
```

However, doing the CString conversion manually works:
```rust
            Err(err) => (error_t::invalid_result_data, result_details_t {
                data: CString::new(err.to_string())
                    .expect("Null byte in string_result_t success code doc comments")
                    .into_raw(),
            }),
```

This was referencing the code in the `from` impl here:
https://github.com/LNP-BP/descriptor-wallet/pull/27/files#diff-d3399c793447433124719f22f76f514e89f4bff0280e4675af35ae071b376ee2L179

But that's less-than-ideal, since it's essentially identical code to the From implementation, duplicated over. It's not so simple to implement the error type for CStrings. Anyway, I've pushed that code, let me know if you have a better approach.